### PR TITLE
Build: Add karma-chrome-launcher support 

### PR DESCRIPTION
**Pull request by ossdev07**
Removed phantomjs dependency from package.json and karma.conf.js.

Signed-off-by: ossdev07 <ossdev@puresoftware.com>

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix [template]
[ ] New rule [template]
[X] Changes an existing rule [template]
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

What changes did you make? (Give an overview)
Just added the "customLauncher" support which allow to use HeadlessChrome with --no-sandbox flag. 

Is there anything you'd like reviewers to focus on?
Phantomjs is not supported anymore should use chrome instead.

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -67,8 +67,13 @@ module.exports = function(config) {
 
         // start these browsers
         // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-        browsers: ["PhantomJS"],
-
+        browsers: ["HeadlessChrome"],
+        customLaunchers: {
+          HeadlessChrome: {
+            base: 'ChromeHeadless',
+            flags: [ '--no-sandbox', ],
```
Comment (kaicataldo) [inline karma.conf.js]: Do you mind explaining why this flag is necessary?

**Comment by kaicataldo** [comment]
One more thing: would you mind formatting the commit message following [our guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changes)? We ask this of committers so that we can keep the generation of our changelog automated.

**Comment by not-an-aardvark** [comment]
Note to whoever merges this: Since this just affects the tests and isn't user-facing, the commit prefix should be changed to `Build:` rather than `New:`.

**Comment by Aladdin-ADD** [comment]
Thanks for the PR!

Is something like chrome/puppeteer required?(I'm not seeing it in devdeps).

File: karma.conf.js (hunk 1 above)
Comment (kaicataldo) [reply karma.conf.js]: Leaving this link to the [Travis CI docs](https://docs.travis-ci.com/user/chrome#sandboxing) here for posterity. Thanks for the explanation!

**Comment by not-an-aardvark** [comment]
The commit in this PR was reverted in https://github.com/eslint/eslint/pull/10973 because it was breaking the build on a few platforms. For example, the tests on our Jenkins release server failed with the following error:

```
START:
[33m12 10 2018 15:09:14.220:WARN [watcher]: [39mAll files matched by "/var/lib/jenkins/workspace/Releases/ESLint Release/eslint/node_modules/mocha/mocha.js" were excluded or matched by prior matchers.
[32m12 10 2018 15:09:15.937:INFO [karma]: [39mKarma v3.0.0 server started at http://0.0.0.0:9876/
[32m12 10 2018 15:09:15.938:INFO [launcher]: [39mLaunching browser HeadlessChrome with unlimited concurrency
[32m12 10 2018 15:09:15.945:INFO [launcher]: [39mStarting browser ChromeHeadless
[91m12 10 2018 15:09:15.946:ERROR [launcher]: [39mNo binary for ChromeHeadless browser on your platform.
  Please, set "CHROME_BIN" env variable.

Finished in 0 secs / 0 secs @ 15:09:15 GMT-0400 (Eastern Daylight Time)


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! eslint@5.6.1 ci-release: `node Makefile.js ciRelease`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the eslint@5.6.1 ci-release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

@platinumazure also reported that this change caused the tests to start failing for him locally -- he might be able to give more information about what's going on.

Sorry, I had meant to leave an update on this PR when I reverted the change, but I forgot to do so.

**Comment by ossdev07** [comment]
@not-an-aardvark I understand your concern.
But from the error message which you attached, I can see that npm err is caused because the CHROME_BIN path is not set.
We need to export the chromium binary path for example,
~~~
export CHROME_BIN=/usr/bin/chromium-browser (path to your chromium binary).
~~~
I do understand that there must be some other constraints as well which are causing build to fail as you mentioned @platinumazure has reported, I won't mind looking into the issues which are causing builds to breaks on the specific platforms.
I would really like to see phantomjs removed from eslint.
Thanks again for your support!

**Comment by ossdev07** [comment]
@not-an-aardvark Thanks for asking well there are two ways basically by which we can provide chromium binary:

1.  Install Chromium on your machine for example in ubuntu 
~~~
apt-get install chromium-browser
~~~
After that we need to export path to chromium binary as i mentioned like
~~~
export CHROME_BIN=/usr/bin/chromium-browser (path to our chromium binary).
~~~
2. We can define puppeteer in package.json as an devdependency.Puppeteer download chromium inside node-modules with npm install. We need to set the path to chromium binary inside karma.conf.js.

I will be happy to support if you need me to do any modification in the PR.

**Comment by ossdev07** [comment]
Ping @not-an-aardvark @Aladdin-ADD 

Hey, Sorry to bother you again!
Please, could you tell me what it would take to accept the changes suggested by this PR?
Please do revert if any support is required at my end, I will be happy to help.

**Comment by Aladdin-ADD** [comment]
Hopefully the build failing can be fixed in #11027.

**Comment by ossdev07** [comment]
Thanks for the quick update @Aladdin-ADD

**Pull request merged**